### PR TITLE
refactor: break circular dependencies(-16, +2)

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
introduces specialtxman, which handles validation of special transactions, specialtx is now simply the primitive underlying type. This breaks a lot of the circular depends

Also removes an unneeded `#include <masternode/payments.h>` in net_processing.cpp, which resolves a circular dependency. (we know it's okay to remove b/c masternode/payments.h isn't included in any header files, and removing it doesn't break compilation)

### CODE REVIEW

**Review by PastaPastaPasta — COMMENTED** [review]
test

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/rpc/rpcevo.cpp [hunk 1]
```diff
@@ -291,7 +292,7 @@ static std::string SignAndSendSpecialTx(const CMutableTransaction& tx, bool fSub
     LOCK(cs_main);
 
     CValidationState state;
-    if (!CheckSpecialTx(CTransaction(tx), ::ChainActive().Tip(), state, ::ChainstateActive().CoinsTip())) {
+    if (!WITH_LOCK(cs_main, return CheckSpecialTx(CTransaction(tx), ::ChainActive().Tip(), state, ::ChainstateActive().CoinsTip()))) {
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/rpc/rpcevo.cpp]: agree :) I didn't intend for that to be committed here

**Comment by PastaPastaPasta** [comment]
I made some additional changes, breaking even more circular dependencies

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**